src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -319,8 +319,17 @@ public string RevertCell(long rowId, int columnId)
                 throw new ArgumentOutOfRangeException(nameof(rowId), SR.EditDataUpdateNotPending);
             }
 
+            // Update the row
+            EditRevertCellResult revertResult = pendingEdit.RevertCell(columnId);
+            if (!revertResult.IsRowDirty)
+            {
+                // Make an attempt to remove the clean row edit. If this fails, it'll be handled on commit attempt.
```

Comment:
Perhaps in this comment mention the method where commit handles this. I'm kinda iffy commenting this as code often gets refactored while comments do not, but it could potentially be helpful.